packages/query/src/block-processor.ts

```diff
@@ -141,7 +148,7 @@ export class BlockProcessor implements BlockProcessorInterface {
 
   private async syncAndStore() {
     const fullSyncHeight = await this.indexedDb.getFullSyncHeight();
-    const startHeight = fullSyncHeight ? fullSyncHeight + 1n : 0n;
+    const startHeight = fullSyncHeight !== undefined ? fullSyncHeight + 1n : 0n; // Must compare to undefined as 0n is falsy
```

Comment:
Found an interesting bug. We sync block 0. However, 0 is a falsy value. That means if state was reset between blocks 0 and 1000, we'd have a sct root divergence.

Reply:
Nice catch 👍 
I wonder if this fixes the divergence that occurs on Henry's wallet (I suspect Henry didn't stop synchronization specifically between the 0 and 1000 block)

Reply:
Hmm, I _did_ make transactions in the first 1000 blocks, so if it paused to fetch full tx details, that could have been an opportunity for interruption...